framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -337,54 +335,8 @@ public long getFromValueFromIntRange(AnnotatedTypeMirror atm) {
         if (AnnotationUtils.hasElementValue(anno, "from")) {
             return AnnotationUtils.getElementValue(anno, "from", Long.class, false);
         }
-
-        long from;
         TypeMirror type = atm.getUnderlyingType();
-        switch (type.getKind()) {
-            case INT:
-                from = Integer.MIN_VALUE;
-                break;
-            case SHORT:
-                from = Short.MIN_VALUE;
-                break;
-            case BYTE:
-                from = Byte.MIN_VALUE;
-                break;
-            case CHAR:
-                from = Character.MIN_VALUE;
-                break;
-            case LONG:
-                from = Long.MIN_VALUE;
-                break;
-            case DECLARED:
-                String qualifiedName = TypesUtils.getQualifiedName((DeclaredType) type).toString();
-                switch (qualifiedName) {
-                    case "java.lang.Integer":
-                        from = Integer.MIN_VALUE;
-                        break;
-                    case "java.lang.Short":
-                        from = Short.MIN_VALUE;
-                        break;
-                    case "java.lang.Byte":
-                        from = Byte.MIN_VALUE;
-                        break;
-                    case "java.lang.Character":
-                        from = Character.MIN_VALUE;
-                        break;
-                    case "java.lang.Long":
-                        from = Long.MIN_VALUE;
-                        break;
-                    default:
-                        throw new UserError(
```

Comment:
Previously we got a user error. What happens now when an `@IntRange` is used on something invalid, e.g. a `String`?

Reply:
@smonteiro Could you add a test for this?

Reply:
The original functions to gather from and to values had different BuginCF error messages:
To:
```
throw new UserError( "Illegal type \"@IntRange "	
                                        + qualifiedName	
                                        + "\". @IntRange can be applied to Java integral types.");	
                }	
                break;	
            default:	
                throw new BugInCF(anno.toString() + " on a type of kind " + type.getKind());
```
vs
From:
```
default:	
                        throw new UserError(	
                                "Illegal type \"@IntRange "	
                                        + qualifiedName	
                                        + "\". @IntRange can be applied to Java integral types.");	
                }	
                break;	
            default:	
                throw new BugInCF(	
                        "Tried to apply a default to an IntRange annotation that was neither an integral primitive nor a declared type.");	
        }
``` 
NOTE: that UserError() message seems to be missing the word "only"

The choice of what an unified message should say is up to you guys.
My desire here was to reduce duplication and  reuse existing functions whenever possible.

Reply:
There are three different ways to raise errors: type checking errors, on a particular line; `UserError` which are supposed to be used for invalid flags/configuration; and `BugInCF` for internal errors of the CF.
I would say that the original code here incorrectly used a `UserError`. Instead, this should be a type-checking error raising this problem.

Take the following code:

```java
import org.checkerframework.common.value.qual.IntRange;

class IntRangeUsage {
  @IntRange(to = 0) Object o;
}
```

At the moment we get the following error:

```
javac -processor org.checkerframework.common.value.ValueChecker IntRangeUsage.java
error: Illegal type "@IntRange java.lang.Object". @IntRange can be applied to Java integral types.
1 error
```

That's not particularly useful, because it doesn't point you to the right line in the code.
Changing the `UserError` here to a `BugInCF` will make the error even scarier, because we will think that it is an internal inconsistency and show a stacktrace etc.

So https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java#L285 should itself first validate the underlying java type and then the internal code in `ValueATF` can raise the `BugInCF`.

Could you add the above test and make this change in a separate PR? Then we'll know that this PR doesn't break that.

Reply:
@wmdietl  @mernst 
While researching for that change in the preceding comment, something else seems to have surfaced in that snippet of code.

The definition of a checker framework annotation does not state in a programmatic way what classes the checker annotation is applicable to.
It is implied by naming conventions and javadoc documentation but not enforced as such.

Example on [(2.11.1 live demo )](http://eisop.uwaterloo.ca/live#mode=display) using StringVal as an annotation on an int field.
```
import org.checkerframework.common.value.qual.StringVal;
public class Magician  {
@StringVal({"a","b"}) int i;
@StringVal({"a","b"}) Integer bigI;
}
```
Produces no error message. IMHO, it should.

Enforcing an explicitly declared relationship of "this-annotation-is-applicable-to-these-classes-and-their-subclasses"  seems to in order.

At this time, I cannot tell if the current implementation of the framework checks that relationship for the other known checkers in imperative form in code.

Reply:
@smonteiro I agree that the Constant Value Checker should issue a warning for
```@StringVal({"a","b"}) int i;```
Thanks for pointing that out.

You can declare that an entire checker is only relevant to certain types:
https://checkerframework.org/manual/#creating-relevant-java-types .
I think that annotation should be permitted on a type annotation as well, and should be enforced.

Let's move that discussion to a separate issue, probably https://github.com/typetools/checker-framework/issues/1908.

Reply:
https://github.com/typetools/checker-framework/pull/2942#discussion_r354910201
is being addressed in #2993.
Once that PR is merged into this PR, we can continue reviewing this.